cmstestsuite/testrunner.py

```diff
@@ -304,6 +304,15 @@ def submit_tests(self, concurrent_submit_and_eval=True):
             self.ps.start("Worker", shard)
         if concurrent_submit_and_eval:
             self.ps.start("EvaluationService", contest=self.contest_id)
```

Comment:
I think there is a better arrangement:

<start CWS>
<if concurrent, start ES>
<wait>
<start PS and Workers>
<submit>
<if not concurrent start ES>
<wait>

As you wrote it, if concurrent we wait for workers and PS without the need to, and if not we never wait for PS and workers, which means instead of outputting "cannot start PS" we will output some other unclear error long after the real problem.